### ISSUE TITLE
Previews don't work on a Gatsby Page that isn't a template

### ISSUE THREAD

**Issue by davidli3100** [angeloashmore/gatsby-source-prismic#245]
I'm not sure if this is intentional or not, but previews work on pages that are rendered using the `src/templates/page.js` template, but previews will not work at all if I export `src/pages/index.js` with the preview HOC. 

Is this supposed to happen? Or is there something else I need to do to get previews working in `index.js`? I noticed the example in the docs was done in `src/templates/page.js` so I'm not sure if that is a mandatory requirement or not.

**Comment by seaneking** [comment]
They work fine for me on regular pages as well as template pages FWIW. If you're trying to preview an unpublished page you have to use `withUnpublishedPreview`, and I think that _might_ only work with templates.

**Comment by davidli3100** [comment]
Weird, the page is published. I have the exact opposite issue lol. I can preview unpublished drafts, and other published pages that are built off of a template, but I can't preview any changes on the index page. Weird. I do get an error of `s isn't a function` at `documentToNodes.ts:209`
GitHub <notifications@github.com> wrote:
“They work fine for me on regular pages as well as template pages FWIW. If you're trying to preview an unpublished page you have to use withUnpublishedPreview, and I think that might only work with templates.”

**Comment by davidli3100** [comment]
Weird, the page is published. I have the exact opposite issue lol. I can preview unpublished drafts, and other published pages that are built off of a template, but I can't preview any changes on the index page. Weird. I do get an error of `s isn't a function` at `documentToNodes.ts:209`

**Comment by angeloashmore** [comment]
@davidli3100 Strange, that `isn't a function` error looks like an issue with the link resolver. If it doesn't expose anything sensitive, could you post your `withPreviewResolver` function in your preview page?

**Comment by davidli3100** [comment]
```
export default withPreviewResolver(PreviewPage, {
  repositoryName: "repository",
  linkResolver: ({ node, key, link }) => doc => {
    if (doc.type === "home") return `/`

    return `/${doc.uid}`
  },
})
```

Weird, this is the exact same link resolver used in the docs. It seems to work for other previews? Just not `src/pages/index.js` and I'm assuming that it would break for other static pages if I had them.

**Comment by imjamesdotme** [comment]
Hey @davidli3100 👋 I had a similar issue to the one you've described this week. In case it's helpful and might work for you, here are the details.

I wanted to add previews to a new template and two new pages. I could preview unpublished changes but no preview published ones. After some experimentation and relentlessly trying to figure out why my other template worked, I stumbled across the following. 

The template I had working with previews (both unpublished and published) contained the property `prismicId` in the query but the new template and pages didn't, as soon as I added this property everything worked. Perhaps the preview HOC's rely on this ID when querying. Examples below.

**Before (Only unpublished previews working)**
```
query Page($uid: String) {
    prismicPage(uid: { eq: $uid }) {
      uid
      data {
        title {
          text
        }
        ...
```

**After (Both unpublished & published previews working)**
```
  query Page($uid: String) {
    prismicPage(uid: { eq: $uid }) {
      uid
      prismicId
      data {
        title {
          text
        }
        ...
```

Hope this might help! 👍

**Comment by mjsz** [comment]
fwiw, this lack of the `prismicId` also bit me when trying to figure out why my previews were not reflected edited content. maybe useful to add to the README?

**Comment by imjamesdotme** [comment]
@mjsz  Good call! I've updated the previews documentation and created a PR #266.
